### ISSUE TITLE
unboundlocalerror: local variable 'updated' referenced before assignment eurostatapiclient

### ISSUE THREAD

**Issue by kntrdzs** [opus-42/eurostat-api-client#9]
Hi Emmanuel,

I have been using this package for a while, and until now everything went well. Now I have got an error : 

> unboundlocalerror: local variable 'updated' referenced before assignment

In the dataset.py file, there is a create_json_function where the error comes from. If I add a new line "updated = datetime.now()" at the top of the function, it works, but it always gives back the datetime.now() so the problem can be here:
`updated_raw = json.get("updated")
        date_regex = re.compile(r'\d{4}-\d{1,2}-\d{1,2}')
        if updated_raw and date_regex.match(updated_raw):
            updated = datetime.strptime(
                updated_raw,
                '%Y-%m-%d'
            )
        else:
            datetime.now()`
I think the Eurostat website has changed and this causes the problem. I am trying to figure it out what has changed, but it takes time because I'm not familiar with the Eurostat website.

**Comment by derevirn** [comment]
Hello @kntrdz , thanks for submitting this issue. I have also come across this problem, and I think I found the cause of it. As seen below, the Eurostat v2.1 API returns a null value for the updated variable.

`{"version":"2.0","label":"GDP and main components (output, expenditure and income)","href":"http://ec.europa.eu/eurostat/wdds/rest/data/v2.1/json/en/nama_10_gdp?unit=CP_MEUR&na_item=B1GQ","source":"Eurostat","updated":null,"status": `

I tried out a few things, and realized that changing the version to v2.2 fixes the issue:

`{"version":"2.0","label":"GDP and main components (output, expenditure and income)","href":"http://ec.europa.eu/eurostat/wdds/rest/data/v2.2/json/en/nama_10_gdp?unit=CP_MEUR&na_item=B1GQ","source":"Eurostat","updated":"2022-07-05","status":`

Long story short, you simply need to change the version number on the Eurostat API client object, and it should work fine.

`client = EurostatAPIClient('v2.2', 'json', 'en')`

Let me know if that helps, cheers!

**Comment by opus-42** [comment]
Hello, I would suggest to close that @kntrdzs and @derevirn, I would suggest to close that issue has it seems that you have found a solution and there is no patch needed. Let me know if I have misunderstood

Best

**Comment by derevirn** [comment]
@opus-42 It might have been a temporary bug, as I can't replicate it by using v2.1 of the API. Still, the problem originated from Eurostat, not the library, so I think you can close the issue.
